testapi.pm

```diff
@@ -1477,6 +1483,42 @@ sub mouse_set ($mx, $my) {
     query_isotovideo('backend_mouse_set', {x => $mx, y => $my});
 }
 
+=head2 mouse_move
+
+  mouse_move($x, $y, [$pixels]);
+
+Slowly move mouse pointer to given coordinates
+
+=cut
+
+sub mouse_move ($x, $y, $pixels = 5) {
+    my $old = query_isotovideo('backend_get_last_mouse_set');
+    my $curx = $old->{x};
+    my $cury = $old->{y};
+    return if $curx < 0 or $cury < 0;
+    while (1) {
+        my $diffx = $x - $curx;
+        my $diffy = $y - $cury;
+        if (abs($diffx) < $pixels) {
+            $curx = $x;
+        }
+        else {
+            $curx += $diffx < 0 ? -$pixels : +$pixels;
+        }
+        if (abs($diffy) < $pixels) {
+            $cury = $y;
+        }
+        else {
+            $cury += $diffy < 0 ? -$pixels : +$pixels;
+        }
```

Comment:
I think you want to use the `clamp` function here. It looks like Perl doesn't have such a function but you can find a possible implementation in C++ here: https://en.cppreference.com/w/cpp/algorithm/clamp

Then it would just be `$curx += clamp($x - $curx, -$pixels, +$pixels)` and the same for y.

Reply:
I've just translated the C++ implementation to Perl:

```
sub clamp($v, $lo, $hi) { $v < $lo ? $lo : $hi < $v ? $hi : $v }
```

Reply:
Thanks, that works :)